### DISCUSSION TITLE
Double inner product of two RankTwotensor

### DISCUSSION THREAD

**Discussion by Bigdogcat** [idaholab/moose#24822] [Q&A Modules: Solid mechanics] [answered]
Hi moose community,
I would like to ask what's the command for the double inner product of two RankTwotensor.

Here I want to calculate the plastic work increment _plastic_work_incr = _stress_old[_qp] : _plastic_strain_incr[_qp], where _plastic_work_incr is a scalar and _stress_old and _plastic_strain_incr are RankTwotensor.

Thanks very much!

**Comment by hugary1995** [comment, marked as answer]
` _stress_old[_qp].doubleContraction(_plastic_strain_incr[_qp])`

**Reply by Bigdogcat** [reply]
Thanks!
